### PR TITLE
Use the lsan public interface in docs and tests

### PR DESCRIPTION
**Pull request by tlively**
Patch originally by @waywardmonkeys in #10938.

### CODE REVIEW

**Comment by RReverser** [comment]
Doesn't matter, but just FYI as maintainer you should be able to just push commits to the original PR / branch instead of having to create a new one :)

**Comment by RReverser** [comment]
Actually, is there any difference between this PR and original? Hard to spot.

File: tests/pthread/test_pthread_lsan_no_leak.cpp [hunk 1]
```diff
@@ -15,9 +15,7 @@ void g(void) {
   while (1);
 }
 
-namespace __lsan {
-  void DoLeakCheck();
-}
+extern "C" void __lsan_do_leak_check(void);
```
Comment (sbc100) [inline tests/pthread/test_pthread_lsan_no_leak.cpp:18]: You should be able to get this via a header now, right?

**Comment by tlively** [comment]
> Doesn't matter, but just FYI as maintainer you should be able to just push commits to the original PR / branch instead of having to create a new one :)

I don't think I'd be able to push commits to @waywardmonkeys's fork, though.

**Comment by RReverser** [comment]
> I don't think I'd be able to push commits to @waywardmonkeys's fork, though.

AFAIK that's exactly what "allow edits from maintainers" (which is set by default) allows. See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork and https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork.

TL;DR - if there's an open PR, then maintainers are allowed to push to corresponding branch on a fork.

**Comment by RReverser** [comment]
Looks like Travis is still running into https://github.com/emscripten-core/emscripten/pull/10938#issuecomment-621060286. E.g:

https://app.circleci.com/pipelines/github/emscripten-core/emscripten/6546/workflows/25abda6f-519e-4bab-aa81-fcea05d7e838/jobs/332533/steps

**Comment by tlively** [comment]
Ah, too bad about the browser tests. I was hoping they would stop failing when this change was rebased on the latest code. Unfortunately I am not able to run browser tests with my current setup because there is no desktop running on my workstation. Would someone else be willing to look into this for me?

**Comment by sbc100** [comment]
> Ah, too bad about the browser tests. I was hoping they would stop failing when this change was rebased on the latest code. Unfortunately I am not able to run browser tests with my current setup because there is no desktop running on my workstation. Would someone else be willing to look into this for me?
> 
> No, I don't think this is the same postRun issue. The failing tests don't use postRun.

Yeah this is kind of pain.  I generally can't run browser tests either because my machine is remote.  On option is to compile the test and then using `emrun --no_browser` to run a server, then use port forwarding to access it over ssh.  e.g. `ssh -L123:localhost:123 mydesktop`  then browsing to port 123 is the same as if you were browsing from your desktop/server.

**Comment by sbc100** [comment]
Another (better?) option is to move/port those tests out or browser tests since they should run on node too these days.

**Comment by kripken** [comment]
> Would someone else be willing to look into this for me?

It looks like these tests [check for specific line numbers](https://github.com/emscripten-core/emscripten/blob/master/tests/pthread/test_pthread_lsan_leak.js#L8) which this patch changes. I guess you can either update those files, or somehow avoid changing line numbers by adding/removing lines :wink:

**Comment by tlively** [comment]
Thanks! I'll try blindly changing the line numbers for now, but if that doesn't work I'll put in more effort to try to port these tests to node instead of the browser.
